packages/testlab/src/client.ts

```diff
@@ -26,22 +26,33 @@ export function createClientForHandler(
   return supertest(server);
 }
 
+export interface RestServer {
+  options?: {
+    http?: {
+      port?: number;
+    },
+  };
+  handleHttp(req: http.ServerRequest, res: http.ServerResponse): void;
+}
+
 export interface Server {
-  config: {port: number};
-  start(): Promise<void>;
+  start() : Promise<void>;
+  options?: {
+    // tslint:disable-next-line:no-any
+    [prop: string] : any,
+  };
 }
 
-export async function createClientForServer(server: Server): Promise<Client> {
+export async function createClientForRestServer(server: Server):
+  Promise<Client> {
   await server.start();
-  const url = `http://127.0.0.1:${server.config.port}`;
+  const port = server.options && server.options.http
+    ? server.options.http.port : 3000;
+  const url = `http://127.0.0.1:${port}`;
   // TODO(bajtos) Find a way how to stop the server after all tests are done
   return supertest(url);
 }
 
-export interface Application {
-  handleHttp(req: http.ServerRequest, res: http.ServerResponse): void;
-}
-
-export function createClientForApp(app: Application): Client {
-  return createClientForHandler(app.handleHttp);
+export function createClientForServer(server: RestServer) {
```

Comment:
We have two similarly named helpers: `createClientForRestServer` and `createClientForServer`. AFAICT, the first starts the server via `server.start()`, the second passes `server.handleHttp` function to supertest and lets supertest to take care of starting/stopping the HTTP server.

I am proposing to either unify these helpers into a single one, or else choose different names to make the difference more clear.

I also think we should keep some sort of a helper function for creating supertest client for a given application, because acceptance test should be end-to-end and call `app.start()` to start any servers configured in the app.

Reply:
I've removed `createClientForServer` and refactored the code to make use of `createClientForHandler` where it was previously used.

I think writing the application-level client utility function would be premature given that nothing currently needs it. If the requirement comes up in the future, then whomever is creating the code with that need should provide that utility.